nexus/src/app/background/tasks/vpc_routes.rs

```diff
@@ -202,12 +205,65 @@ impl BackgroundTask for VpcRouteManager {
 
                 // resolve into known_rules on an as-needed basis.
                 for set in &route_sets {
-                    let Some(db_router) = db_routers.get(&set.id) else {
-                        // The sled wants to know about rules for a VPC
-                        // subnet with no custom router set. Send them
-                        // the empty list, and unset its table version.
-                        set_rules(set.id, None, HashSet::new());
-                        continue;
+                    let (db_router, igw_only) = match db_routers.get(&set.id) {
+                        Some(r) => (r.clone(), false),
+                        None => {
+                            info!(
+                                log,
+                                "VPC route manager no upstream router\nkey: {:#?}\nkeys: {:#?}",
+                                set.id,
+                                db_routers.keys();
+                                "sled" =>  sled.id().to_string(),
+                            );
+
+                            // TODO the following is working around the fact that
```

Comment:
Calling explicit attention to this comment.

Reply:
I believe this is still needed in order to get custom IGW routes down to OPTE. However, this code seems to be making the `test_vpc_routers_custom_delivered_to_instance` test flaky.

Reply:
Removed in favor or attaching IGW routes to subnets through the API.